eclipse-theia/theia
# notebook open optimizations

#### What it does

One larger performance issue for notebooks was the creation of the `MonacoEditorModels` used by the cell editors. The problem was that they needed to be created when the notebook is first loaded to be available to the extensions but where created one by one which required firing a network request for each single `MonacoEditorModel`. For a notebook with 100 cells this took up to 500ms.

This PR changes that so that the documents are created seperately between backend and frontend. 
In the backend they are created when the document for the notebook itself is created. In the frontend they are created afterwards in `createNotebookModel` in the `notebook-service.ts` 

Im not sure how nice the solution with rebinding the `MonacoTextModelService` is. Would love some feedback on this

#### How to test

Since its about performance, best way to test is to check if working with notebooks is still working.
open, execute cell, change cell, execute again

#### Follow-ups



#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
